server.go

```diff
@@ -127,7 +131,7 @@ func (s *RPCServer) DeregisterHandler(rpc, topic string) {
 	h, ok := s.handlers[key]
 	s.mu.RUnlock()
 	if ok {
-		go h.close()
+		h.close(true)
```

Comment:
we probably want this to be `false`

Reply:
when a caller deregisters a handler it's only after they have no further interest in the resource it is serving requests for. typically this happens after the caller has synchronized shutdown with any interested parties. the only context where shutting down gracefully makes sense imo is when the entire server is going away

Reply:
I think we need to find another way around this. Egress will, for example, deregister itself from the StartEgress topic when it doesn't have enough CPU to accept a request, but that doesn't mean it should discard any currently active requests